### PR TITLE
Migrate `extern-diff-internal-name`, `extern-multiple-copies` and `extern-multiple-copies2` `run-make` tests to rmake

### PR DESCRIPTION
**Pull request by Oneirical**
Part of #121876 and the associated [Google Summer of Code project](https://blog.rust-lang.org/2024/05/01/gsoc-2024-selected-projects.html).

Please try:

try-job: test-various

### CODE REVIEW

**Review by Kobzol — COMMENTED** [review]
LGTM if try job passes.

File: tests/run-make/extern-diff-internal-name/rmake.rs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// In the following scenario:
+// 1. The crate foo, is referenced multiple times
+// 2. --extern foo=./path/to/libbar.rlib is specified to rustc
+// 3. The internal crate name of libbar.rlib is not foo
+// Compilation fails with the "multiple crate versions" error message.
+// As this was fixed in #17189, this regression test ensures this bug does not
+// make a resurgence.
+// See https://github.com/rust-lang/rust/pull/17189
+
+//FIXME(Oneirical): test-various
```
Comment (jieyouxu) [inline tests/run-make/extern-diff-internal-name/rmake.rs:10]: Remark: left over FIXME

File: tests/run-make/extern-multiple-copies/rmake.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// In this test, the rust library foo1 exists in two different locations, but only one
+// is required by the --extern flag. This test checks that the copy is ignored (as --extern
+// demands fetching only the original instance of foo1) and that no error is emitted, resulting
+// in successful compilation.
+// https://github.com/rust-lang/rust/pull/29961
+
+// FIXME(Oneirical): test-various
+
+use run_make_support::{path, rfs, rust_lib_name, rustc};
```
Comment (jieyouxu) [inline tests/run-make/extern-multiple-copies/rmake.rs:9]: Remark: and here

File: tests/run-make/extern-multiple-copies/rmake.rs [hunk 3]
```diff
@@ -0,0 +1,17 @@
+// In this test, the rust library foo1 exists in two different locations, but only one
+// is required by the --extern flag. This test checks that the copy is ignored (as --extern
+// demands fetching only the original instance of foo1) and that no error is emitted, resulting
+// in successful compilation.
+// https://github.com/rust-lang/rust/pull/29961
+
+// FIXME(Oneirical): test-various
```
Comment (jieyouxu) [inline tests/run-make/extern-multiple-copies/rmake.rs:7]: Remark: leftover FIXME

File: tests/run-make/extern-multiple-copies2/rmake.rs [hunk 4]
```diff
@@ -0,0 +1,24 @@
+// Almost identical to `extern-multiple-copies`, but with a variation in the --extern calls
+// and the addition of #[macro_use] in the rust code files, which used to break --extern
+// until #33625.
+// In this test, the rust library foo1 exists in two different locations, but only one
+// is required by the --extern flag. This test checks that the copy is ignored (as --extern
+// demands fetching only the original instance of foo1) and that no error is emitted, resulting
+// in successful compilation.
+// https://github.com/rust-lang/rust/issues/33762
+
+// FIXME(Oneirical): test-various
```
Comment (jieyouxu) [inline tests/run-make/extern-multiple-copies2/rmake.rs:10]: Remark: and here

**Comment by Oneirical** [comment]
All ignores removed.

**Comment by Kobzol** [comment]
Thank you! Let's see if this also passes the whole test suite without the ignores.

@bors r+ rollup=iffy

**Pull request merged**